source/model.js

```diff
@@ -52,96 +69,102 @@ function WinnerChecker(field){
                 }
             }
             if (isWinnerFound) {
-                return currentWinCombination = winArray[winSegment];
+                return winArray[winSegment];
             }
         }
         return false;
-    }
-}
+    };
+};
 
-function Player(name, symbol, field){
-    let gameField = field;
+function Player(name, symbol){
     this.playerName = name;
     this.playerSymbol = symbol;
-    this.myGame = null;
 
-    this.startMove = function(){ };
+    this.startMove = function(game){ };
     this.finishMove = function (cellNum){
-        return gameField.updateCell(cellNum, this.playerSymbol);
+        return singletonField.getInstance().updateCell(cellNum, this.playerSymbol);
     };
-}
+};
 
-function ComputerPlayer(name, symbol, field){
+function ComputerPlayer(name, symbol){
     Player.apply(this, arguments);
 
-    this.startMove = function(){
+    this.startMove = function(game){
         let computerCellNum = cellFinder();
-        this.myGame.finishTurn(computerCellNum);
+        game.finishTurn(computerCellNum);
     };
 
     function cellFinder(){
-        let emptyCellsArr = field.getEmptyCells();
+        let emptyCellsArr = singletonField.getInstance().getEmptyCells();
         let min = 0;
         let max = emptyCellsArr.length > 0 ? emptyCellsArr.length - 1 : 0;
         let random = Math.floor(Math.random() * (max - min + 1)) + min;
         return emptyCellsArr[random];
-    }
+    };
 }
 
 ComputerPlayer.prototype = Object.create(Player.prototype);
 ComputerPlayer.prototype.constructor = ComputerPlayer;
 
-function Game(humanPlayer, firstPlayerName, secondPlayerName){
+function Iterator(array) {
+    let index = 0;
+
+    this.getCurrent = function(){
+        return array[index];
+    };
+
+    this.moveNext = function(){
+        index = index == array.length-1 ? 0 : ++index;
+    };
+};
+
+function Game(players){
     let self = this;
-    let currentPlayerIndex = 0;
     let gameStatuses = ['Playing', 'Winner', 'Tie'];
-    let field = new Field();
+    let field = singletonField.getInstance();
     let winnerChecker = new WinnerChecker(field);
-    let players = playerCreator();
+    let iterator = new Iterator(players);
 
-    this.currentPlayer = players[currentPlayerIndex];
     this.currentStatus = gameStatuses[0];
 
+    this.currentPlayerName = function(){
```

Comment:
Can we rename this function so that its name starts with a verb (getCurrentPlayerName)? Currently the name of this function may make people think that this is a property.